spring-rabbit-ext/src/main/java/org/springframework/amqp/rabbit/log4j2/AmqpAppender.java

```diff
@@ -0,0 +1,495 @@
+
+
+	public AmqpAppender(String name, Filter filter, Layout<? extends Serializable> layout, boolean ignoreExceptions,
+			AmqpManager manager) {
+		super(name, filter, layout, ignoreExceptions);
+		this.manager = manager;
+	}
+
+	@PluginFactory
+	public static AmqpAppender createAppender(
+			@PluginAttribute("name") String name,
+			@PluginElement("Layout") Layout<? extends Serializable> layout,
+			@PluginElement("Filter") Filter filter,
+			@PluginAttribute("ignoreExceptions") boolean ignoreExceptions,
+			@PluginAttribute("host") String host,
+			@PluginAttribute("port") int port,
+			@PluginAttribute("user") String user,
+			@PluginAttribute("password") String password,
+			@PluginAttribute("virtualHost") String virtualHost,
+			@PluginAttribute("senderPoolSize") int senderPoolSize,
+			@PluginAttribute("maxSenderRetries") int maxSenderRetries,
+			@PluginAttribute("applicationId") String applicationId,
+			@PluginAttribute("routingKeyPattern") String routingKeyPattern,
+			@PluginAttribute("generateId") boolean generateId,
+			@PluginAttribute("deliveryMode") String deliveryMode,
+			@PluginAttribute("exchange") String exchange,
+			@PluginAttribute("exchangeType") String exchangeType,
+			@PluginAttribute("declareExchange") boolean declareExchange,
+			@PluginAttribute("durable") boolean durable,
+			@PluginAttribute("autoDelete") boolean autoDelete,
+			@PluginAttribute("contentType") String contentType,
+			@PluginAttribute("contentEncoding") String contentEncoding,
+			@PluginAttribute("charset") String charset) {
+		if (name == null) {
+			LogFactory.getLog("log4j2AppenderErrors").error("No name for AmqpAppender");
+		}
+		Layout<? extends Serializable> theLayout = layout;
+		if (theLayout == null) {
+			theLayout = PatternLayout.createDefaultLayout();
+		}
+		AmqpManager manager = new AmqpManager(name);
+		manager.host = host;
+		manager.port = port;
+		manager.username = user;
+		manager.password = password;
+		manager.virtualHost = virtualHost;
+		manager.senderPoolSize = senderPoolSize;
+		manager.maxSenderRetries = maxSenderRetries;
+		manager.applicationId = applicationId;
+		manager.routingKeyPattern = routingKeyPattern;
+		manager.generateId = generateId;
+		manager.deliveryMode = MessageDeliveryMode.valueOf(deliveryMode);
+		manager.exchangeName = exchange;
+		manager.exchangeType = exchangeType;
+		manager.declareExchange = declareExchange;
+		manager.durable = durable;
+		manager.autoDelete = autoDelete;
+		manager.contentType = contentType;
+		manager.contentEncoding = contentEncoding;
+		manager.charset = charset;
+		AmqpAppender appender = new AmqpAppender(name, filter, theLayout, ignoreExceptions, manager);
+		manager.activateOptions();
+		appender.startSenders();
+		return appender;
+	}
+
+	/**
+	 * Submit the required number of senders into the pool.
+	 */
+	private void startSenders() {
+		for (int i = 0; i < manager.senderPoolSize; i++) {
+			manager.senderPool.submit(new EventSender());
```

Comment:
I wonder if we can avoid this in favor of built-in async feature: https://logging.apache.org/log4j/2.x/manual/async.html